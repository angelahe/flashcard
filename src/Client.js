// interract with db back end

function createDeck() {
  return fetch('/api/deck', {
    method: 'POST',
  })
    .then(response => response.json())
    .then(json => json.id);
}

function createCard(L1word, L2word) {
  // let's try to send an api call with hardcoded values like in postman
  // return fetch('api/card?EN_word=test&ES_word=prueba', {
  // let's try to send the values passed in
  return fetch(`api/card?EN_word=${L1word}&ES_word=${L2word}`,
    {
    // return fetch('/api/card', {
      method: 'POST',
    })
    .then(response => response.json())
    .then(json => json.id);
}
const Client = { createDeck, createCard };
export default Client;
