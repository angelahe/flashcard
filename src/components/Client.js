// interract with db back end

function createDeck() {
  return fetch('/api/deck', {
    method: 'POST',
  })
    .then(response => response.json())
    .then(json => json.id);
}

function createCard(deckId, l1word, l2word) {
  // let's try to send an api call with hardcoded values like in postman
  // return fetch('api/card?EN_word=test&ES_word=prueba', {
  // let's try to send the values passed in
  // 127.0.0.1:5000/api/deck/68dfd811-9913-422f-8c50-037055314e13/card?EN_word=test&ES_word=prueba
  console.log(`args are deck:${deckId}, l1word: ${l1word}, l2word: ${l2word}!`);
  return fetch(`api/deck/${deckId}/card?EN_word=${l1word}&ES_word=${l2word}`,
  //  return fetch(`api/deck/card?EN_word=${l1word}&ES_word=${l2word}`,
    {
    // return fetch('/api/card', {
      method: 'POST',
    })
    .then(response => response.json())
    .then(json => json.id);
}
const Client = { createDeck, createCard };
export default Client;
