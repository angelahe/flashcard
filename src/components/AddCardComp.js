import React from 'react';
import Client from './Client';

class AddCardComp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentCard: '',
      currentL1: '',
      currentL2: '',
    };
  }

  handleL1Change = (event) => {
    this.setState({ currentL1: event.target.value });
  };

  handleL2Change = (event) => {
    this.setState({ currentL2: event.target.value });
  };

  handleAddCardClick = () => {
    const { deck, addCard } = this.props;
    const { currentL1, currentL2 } = this.state;
    console.log('deck in handleAddCardClick is: ', deck);

    Client.createCard(deck, currentL1, currentL2)
      .then(id => this.setState({ currentCard: id },
        // eslint-disable-next-line react/destructuring-assignment
        () => {
          addCard(this.state.currentCard)
          console.log('the card that was assigned is ', this.state.currentCard)
        }));
  };

  render() {
    const { currentCard, currentL1, currentL2 } = this.state;
    return (
      <div>
        <h1>Add a Card to the Deck</h1>
        <span>L1 word:</span>
        <input value={currentL1} onChange={this.handleL1Change} id="l1word" />
        <br />
        <span>L2 word:</span>
        <input value={currentL2} onChange={this.handleL2Change} id="l2word" />
        <br />
        <button type="button" onClick={this.handleAddCardClick}>
          Add Card
        </button>
        {(currentCard !== '') ? <p>current card is: {currentCard}</p> : null}
      </div>
    );
  }
}
export default AddCardComp;
