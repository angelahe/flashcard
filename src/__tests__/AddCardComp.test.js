import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import AddCardComp from '../components/AddDeckComp';
import Client from '../components/Client';

jest.mock('../components/Client');

it('renders without crashing', () => {
  shallow(<AddCardComp />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AddCardComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('AddCardComp renders correctly', () => {
  const tree = renderer.create(<AddCardComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('does not show the current card when there is no card', () => {
  const component = shallow(<AddCardComp />);
  expect(component.html()).not.toContain('current card');
});

it('shows the current card when there is one', () => {
  const cardId = '12345-ABCD';
  const component = shallow(<AddCardComp />);
  component.setState({ currentCard: cardId });
  component.update();

  expect(component.html()).toContain(`current card: ${cardId}`);
});

it('should set the state when createCard completes', (done) => {
  const cardId = '67890-EFGH';
  Client.createCard = jest.fn(() => Promise.resolve(cardId));

  const component = shallow(<AddCardComp />);
  const button = component.find('button').first();
  button.simulate('click');

  setTimeout(() => {
    expect(component.instance().state.currentCard).toBe(cardId);
    done();
  });
});
