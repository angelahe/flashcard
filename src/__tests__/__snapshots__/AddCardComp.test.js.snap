// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCardComp renders correctly 1`] = `
<div>
  <div>
    <p>
      Add a Deck
    </p>
    <button
      onClick={[Function]}
      type="button"
    >
      Add Deck
    </button>
  </div>
</div>
`;
